pkg/relayer/proxy/sync.go

```diff
@@ -37,7 +42,12 @@ func (server *relayMinerHTTPServer) serveSyncRequest(
 		logger.Warn().Err(err).Msg("failed creating relay request")
 		return relayRequest, err
 	}
-	request.Body.Close()
+	defer func(Body io.ReadCloser) {
```

Comment:
Body should already be closed at this point, since `server.newRelayRequest` might return an error.

Could you create a `closeBody` method
```go
func closeBody(body io.ReadCloser, logger polylog.Logger)
```
Use it in `newRelayRequest`
```go
func (sync *relayMinerHTTPServer) newRelayRequest(request *http.Request) (*types.RelayRequest, error) {
	requestBody, err := io.ReadAll(request.Body)
	defer closeBody(request.Body, sync.logger)
	...
}